crates/kubelet/src/server.rs

```diff
@@ -68,15 +68,46 @@ fn get_ping() -> Response<Body> {
 /// Implements the kubelet path /containerLogs/{namespace}/{pod}/{container}
 async fn get_container_logs<T: Provider + Sync>(
     provider: &T,
-    _req: &Request<Body>,
+    req: &Request<Body>,
 ) -> Response<Body> {
-    // TODO: extract the right data from the request.
+    // Basic validation steps
+    let path: Vec<&str> = req.uri().path().split('/').collect();
+    // Because of the leading slash, index 0 is an empty string. Index 1 is the
+    // container logs path
+    if path.len() != 5 {
+        let mut res = Response::new(Body::from(format!(
```

Comment:
Should we do this instead:

```
return Response::builder().status(StatusCode::NOT_FOUND).body(...);
```

Reply:
Most definitely. I was just copy pasting and didn’t even think of looking